Album/ViewControllers/ProfileControllers/ProfileViewController.m

```diff
@@ -35,31 +38,56 @@ - (void)viewDidLoad {
+    self.colorConvertHelper = [[ColorConvertHelper alloc] init];
     // Store current user
     self.currentUser = [PFUser currentUser];
     // Assign collection view delegate and dataSource
     self.friendsCollectionView.delegate = self;
     self.friendsCollectionView.dataSource = self;
-    // Fetch friends
-    [self.apiHelper fetchFriends:self.currentUser.objectId withBlock:^(NSArray *friendArr, NSError *error) {
-        if (friendArr != nil) {
-            self.friendsArray = friendArr;
-            dispatch_async(dispatch_get_main_queue(), ^{
-                [self.friendsCollectionView reloadData];
-            });
-        } else {
-            NSLog(@"%@", error.localizedDescription);
-        }
-    }];
-    // Load profile image
-    [self setProfile];
-} /* viewDidLoad */
+    
+}
 
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
-    [self.friendsCollectionView reloadData];
+    self.currentUser = [PFUser currentUser];
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        self.overlayView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, [UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height)];
+
+        self.overlayView.backgroundColor = [UIColor whiteColor];
+        self.overlayView.alpha = 1;
+        UIActivityIndicatorView *activityView = [[UIActivityIndicatorView alloc] init];
+        activityView.center = self.view.center;
+        [self.overlayView addSubview:activityView];
+        [activityView startAnimating];
+        [self.view addSubview:self.overlayView];
+        [self.view bringSubviewToFront:self.overlayView];
+        // Fetch friends
+        [ParseAPIHelper fetchFriends:self.currentUser.objectId withBlock:^(NSArray *friendArr, NSError *error) {
+            if (friendArr != nil) {
+                self.friendsArray = friendArr;
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [self.friendsCollectionView.layer setCornerRadius:15];
+                    [self.friendsCollectionView.layer setShadowRadius:5];
+                    [self.friendsCollectionView.layer setShadowColor:[[UIColor grayColor] CGColor]];
+                    [self.friendsCollectionView.layer setShadowOpacity:1];
+                    [self.friendsCollectionView.layer setShadowOffset:CGSizeMake(0,0)];
+                    self.friendsCollectionView.layer.masksToBounds = NO;
+                    self.friendsCollectionView.clipsToBounds = NO;
+                    [self.friendsCollectionView reloadData];
+                    // Load profile image
+                    [self setProfile];
+                    [UIView transitionWithView:self.view duration:2 options:UIViewAnimationOptionTransitionNone animations:^(void){self.overlayView .alpha=0.0f;} completion:^(BOOL finished){[self.overlayView  removeFromSuperview];}];
+                });
+            } else {
+                NSLog(@"%@", error.localizedDescription);
```

Comment:
lets show an alert, we should have a static helper function for alerts

Reply:
Done!